Lib/test/test_tabnanny.py

```diff
@@ -0,0 +1,353 @@
+        """
+        with TemporaryDirectory() as tmp_dir:
+            lines = ["%r: listing directory\n" % (tmp_dir,),]
+            for i in range(2):
+                file_path = create_python_file(
+                    SOURCE_CODES["error_free"], dir=tmp_dir
+                )
+                lines.append("%r: Clean bill of health.\n" % (file_path,))
+
+            tabnanny.verbose = 1
+            with captured_stdout() as stdout, captured_stderr() as stderr:
+                tabnanny.check(tmp_dir)
+            stdout = stdout.getvalue()
+            for line in lines:
+                self.assertIn(line, stdout)
+            self.assertEqual(stderr.getvalue(), "")
+
+    def test_correct_directory(self):
+        """Directory which contains few error free python source code files."""
+        with TemporaryDirectory() as tmp_dir:
+            for i in range(2):
+                create_python_file(SOURCE_CODES["error_free"], dir=tmp_dir)
+            self._verify_tabnanny_check(tmp_dir)
+
+    def test_when_wrong_indented(self):
+        """A python source code file eligible for raising `IndentationError`."""
+        file_path = create_python_file(SOURCE_CODES["wrong_indented"])
+        msg = ('unindent does not match any outer indentation level'
+                    ' (<tokenize>, line 3)\n')
+        msg = "%r: Indentation Error: %s" % (file_path, msg)
+        self._verify_tabnanny_check(file_path, err=msg)
+
+    def test_when_tokenize_tokenerror(self):
+        """A python source code file eligible for raising 'tokenize.TokenError'.
+        """
+        file_path = create_python_file(SOURCE_CODES["incomplete_expression"])
+        msg = "('EOF in multi-line statement', (7, 0))\n"
+        msg = "%r: Token Error: %s" % (file_path, msg)
+        self._verify_tabnanny_check(file_path, err=msg)
+
+    def test_when_nannynag_error_verbose(self):
+        """A python source code file eligible for raising `tabnanny.NannyNag`.
+
+        Tests will assert `stdout` after activating `tabnanny.verbose` mode.
+        """
+        file_path = create_python_file(SOURCE_CODES["nannynag_errored"])
+
+        out = "%r: *** Line 3: trouble in tab city! ***\n" % (file_path,)
+        out += "offending line: '\\tprint(\"world\")\\n'\n"
+        out += "indent not equal e.g. at tab size 1\n"
+
+        tabnanny.verbose = 1
+        self._verify_tabnanny_check(file_path, out=out)
+
+    def test_when_nannynag_error(self):
+        """A python source code file eligible for raising `tabnanny.NannyNag`.
+        """
+        file_path = create_python_file(SOURCE_CODES["nannynag_errored"])
+        msg = "%s 3 '\\tprint(\"world\")\\n'\n" % (file_path,)
+        self._verify_tabnanny_check(file_path, out=msg)
+
+    def test_when_no_file(self):
+        """A python file which does not exists actually in system."""
+        file_path = 'no_file.py'
+        msg = "%r: I/O Error: [Errno 2] No such file or directory: %r\n" % (
```

Comment:
I am confused for this. There are code samples which are used as input to more than one class. Because of that reason, I believe it is good to keep one module level non-repeated copy of them. It will be more effective if we want to update particular situation of copy of code.